modules/beta-autopilot-public-cluster/outputs.tf

```diff
@@ -24,6 +24,16 @@ output "cluster_id" {
 output "name" {
   description = "Cluster name"
   value       = local.cluster_name
+  depends_on = [
+    /* Nominally, the cluster name is populated as soon as it is known to Terraform.
+    * However, the cluster may not be in a usable state yet.  Therefore any
+    * resources dependent on the cluster being up will fail to deploy.  With
+    * this explicit dependency, dependent resources can wait for the cluster
+    * to be up.
+    */
+    google_container_cluster.primary,
+    google_container_node_pool.pools,
```

Comment:
Autopilot wont have pools, so you will need to add a conditional in autogen. Example
https://github.com/terraform-google-modules/terraform-google-kubernetes-engine/blob/9f06ef4562adcc9096fbf2da6041746d3d08c483/autogen/main/outputs.tf.tmpl#L60-L63

Reply:
Done, thanks